rbac/providers/common/common.py

```diff
@@ -31,11 +42,44 @@ def save_sync_time(sync_source, sync_type):
     r.table("sync_tracker").insert(sync_entry).run()
 
 
-def check_for_sync(source, sync_type):
-    """Check to see if a sync has occurred and return payload."""
-    return (
-        r.table("sync_tracker")
-        .filter({"source": source, "sync_type": sync_type})
-        .coerce_to("array")
-        .run()
-    )
+def get_last_sync(source, sync_type):
+    """
+        Search and get last sync entry from the specified source. Throws
+        ExpectedError if sync_tracker table has not been initialized.
+    """
+    try:
+        last_sync = (
+            r.table("sync_tracker")
+            .filter({"source": source, "sync_type": sync_type})
+            .coerce_to("array")
+            .run()
+        )
+        return last_sync
+    except (r.ReqlOpFailedError, r.ReqlDriverError) as err:
+        raise ExpectedError(err)
+    except Exception as err:
+        LOGGER.warning(type(err).__name__)
+        raise err
+
+
+def check_last_sync(sync_source, sync_type):
+    """
+        Check to see if a sync has occurred and return payload. If the
+        the sync_tracker table is not initialized, this function will
+        keep checking until the table has been initialized.
+    """
+    LOGGER.debug("Checking for previous %s initial sync...", sync_source)
+    while True:
+        try:
+            db_payload = get_last_sync(sync_source, sync_type)
+            return db_payload
+        except ExpectedError:
+            LOGGER.debug(
+                "Sync tracker table has not been initialized. Checking again in %s seconds",
+                DELAY,
+            )
+            time.sleep(DELAY)
+            continue
+        except Exception as err:
```

Comment:
Not sure this block is needed since it ultimately raises, but no major issue in having it